### PR TITLE
[Apps] Support multiple external components per app

### PR DESCRIPTION
**Pull request by lolimay**
depends on https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/150

### CODE REVIEW

File: app/apps/server/orchestrator.js [hunk 1]
```diff
@@ -82,9 +82,10 @@ class AppServerOrchestrator {
 		return this._manager;
 	}
 
-	getEnabledGames() {
-		const { externalComponentManager: { providedComponents } } = this._manager;
-		return providedComponents;
+	getRegisteredExternalComponents() {
+		const { externalComponentManager: { registeredExternalComponents } } = this._manager;
```
Comment (graywolf336) [inline app/apps/server/orchestrator.js]: Move this to a method call. `this._manager.getExternalComponentManager().getProvidedComponents()`

File: app/apps/server/communication/rest.js [hunk 2]
```diff
@@ -249,11 +249,18 @@ export class AppsRestApi {
...
 
-		this.api.addRoute('games', { authRequired: false }, {
+		this.api.addRoute('externalComponents', { authRequired: false }, {
 			get() {
-				const games = orchestrator.getEnabledGames();
+				const registeredExternalComponents = orchestrator.getRegisteredExternalComponents();
+				const externalComponents = [];
 
-				return API.v1.success({ games: Array.from(games.values()) });
+				registeredExternalComponents.forEach((appExternalComponents) => {
```
Comment (graywolf336) [inline app/apps/server/communication/rest.js]: Move this logic inside of the Engine code base and into a new method that only returns an array of external components which are enabled and ready for usage.

File: app/apps/server/orchestrator.js (hunk 1 above)
Comment (lolimay) [reply app/apps/server/orchestrator.js]: Resolved in https://github.com/RocketChat/Rocket.Chat/pull/15495/commits/78e4b264551c30da41bd8f1c3a4c363c03d9f994

File: app/apps/server/communication/rest.js (hunk 2 above)
Comment (lolimay) [reply app/apps/server/communication/rest.js]: Resolved in https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/150/commits/34b5a9b95ec5acff2befaa781567abedf57c88a3

**Pull request merged**